lib/ansible/plugins/loader.py

```diff
@@ -93,6 +113,61 @@ def add_dirs_to_loader(which_loader, paths):
         loader.add_directory(path, with_subdir=True)
 
 
+class PluginLoadContext(object):
+    def __init__(self):
+        self.original_name = None
+        self.redirect_list = []
+        self.error_list = []
+        self.import_error_list = []
+        self.load_attempts = []
+        self.pending_redirect = None
+        self.exit_reason = None
+        self.plugin_resolved_path = None
+        self.plugin_resolved_name = None
+        self.deprecated = False
+        self.removal_date = None
+        self.deprecation_warnings = []
+        self.resolved = False
+
+    def record_deprecation(self, name, deprecation):
+        if not deprecation:
+            return self
+
+        warning_text = deprecation.get('warning_text', None)
+        removal_date = deprecation.get('removal_date', None)
+        if not warning_text:
+            if removal_date:
```

Comment:
If both `removal_date` and `warning_text` keys are mentioned can the warning_text message be appended to the removal_date message? Currently, the removal_date is ignored if warning_test key is present.
It is usefull in or an entry like:
```
      deprecation:
        removal_date: 2022-06-01
        warning_text: See the plugin documentation for more details.
``` 
Resulting deprecation notice:
`arista.eos.acls has been deprecated and will be removed in a release after 2022-06-01. See the plugin documentation for more details.`